### ISSUE TITLE
Access Denied when Dragging

### ISSUE THREAD

**Issue by Metz01** [meeting-room-booking-system/mrbs-code#3632]
I have a problem with the code. I made a fork for my requirements so the code isn't exactly like the one in the main, but I'm encountering an error. When I try to make a booking selecting a single slot everything works fine, but when I try to do so selecting with the drag it tells me `access denied`. After some research I found that the `showAccessDenied` is called from the file `edit_entry.php` in line `1550`:
``` 
if (!getWritable($create_by, $room_id))
{
  showAccessDenied($view, $view_all, $year, $month, $day, $area, isset($room) ? $room : null);
  exit;
} 
```

Why is this check here? The only modify I did to getWritable is:

``` 
if(isset($auth[$rooms])){
   if(in_array($mrbs_user->username, $auth[$rooms])){
      return true;
    }
    return false;
  }
```

Any idea?

**Comment by Metz01** [comment]
Just tried to remove my modify to `getWritable` and it works, those line of code should give the booking permission of a certain room just to the people I say in the config. But when I use the drag I'm selecting another room, so what's happening?

**Comment by Metz01** [comment]
One more update, I wrote a debug for `$room_id` in the `edit_entry.php`, and it seems that every time that you try to book using drag the `room_id = 1`, instead when selecting a single cell the room id is set correctly

**Comment by campbell-m** [comment]
Just so I understand, when you got Access Denied, is your user allowed to make bookings for the selected rooms?  And not allowed to for room with id 1?

**Comment by campbell-m** [comment]
I suspect the problem is that the code is checking that the user is authorised for what has been the default room, rather than the rooms that have been selected, but I just wanted to understand your case more.

**Comment by Metz01** [comment]
> Just so I understand, when you got Access Denied, is your user allowed to make bookings for the selected rooms? And not allowed to for room with id 1?

Correct, the only room with no access to this user is the first one

**Comment by Metz01** [comment]
Maybe `edit_entry_handler.php` line `719` might have an hint?
```
// If the old sticky room is one of the rooms requested for booking, then don't change the sticky room.
// Otherwise change the sticky room to be one of the new rooms.
if (!in_array($room, $rooms))
{
  $room = $rooms[0];
}
```

**Comment by campbell-m** [comment]
I can see what the problem is.  I'll fix it tomorrow.

**Comment by campbell-m** [comment]
Now fixed in [59a7da4](https://github.com/meeting-room-booking-system/mrbs-code/commit/59a7da42f62aeb3bc9ca2dd4d29b33f1683b3f83).

**Comment by Metz01** [comment]
Awesome, everything works fine now. Thank you as always for your quick response
